2-api/cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const app = express();
+const port = 5000;
+
+app.use(express.json());
+
+const { Pool } = require("pg");
+
+const pool = new Pool({
+  user: "migue",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "Cyf2022",
+  port: 5432,
+});
+
+//Add a new GET endpoint `/products`
+app.get("/products", function(req, res){
+ pool.query(`SELECT product_name, 
+ product_availability.unit_price, 
+ supplier_name FROM 
+ products JOIN  product_availability ON products.id = product_availability.prod_id
+ JOIN suppliers ON supp_id = suppliers.id `)
+ .then((result) => res.json(result.rows))
+     .catch((error) => {
+         console.error(error);
+         res.status(500).json(error);
+     });
+
+
+})
+
+// get customers by id
+app.get("/customers/:id", function (req, res) {
+  const customerId = req.params.id;
+
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
```

Comment:
Using query parameters like this is the right way to do it!

If you did something like this:
`"SELECT * FROM customers WHERE id='" + customerId +"'"`

Then a user could send a request where customer id is for example `';DROP TABLES; SELECT'`, and try to execute SQL statements. This is called a SQL injection attack https://www.w3schools.com/sql/sql_injection.asp

When you use query parameters, the connection pool is clever enough to protect you from this.